beacon_node/lighthouse_network/src/config.rs

```diff
@@ -265,6 +266,18 @@ impl Config {
         }
     }
 
+    /// A helper function to check if the local host has a globally routeable IPv6 address. If so,
+    /// returns true.
+    pub fn is_ipv6_supported() -> bool {
+        // If IPv6 is supported
+        let Ok(std::net::IpAddr::V6(local_ip)) = local_ipv6() else {
```

Comment:
nit: we already depend on [`rtnetlink` ](https://docs.rs/rtnetlink/latest/rtnetlink/) via `rust-libp2p` and `if-watch`.
There's an example on how to do it [here](https://github.com/rust-netlink/rtnetlink/issues/63#issuecomment-2105034192), we could implement it here, or upsteam it as they mention in the discussion

Reply:
Yeah, this is fair. 

It wasn't straight forward to see how to implement this, to ensure just globally routable IP. 

Happy to attempt to use this code snippet, or not import a new library, if its easy, maybe push a commit here, or if its harder we can leave for a future PR?

Reply:
thanks for looking it out age! Yeah I also looked and yeah it's def not trivial, so we can leave for a future PR